MetadataProcessor.Shared/Tables/nanoTypeDefinitionTable.cs

```diff
@@ -155,11 +173,50 @@ protected override void WriteSingleItem(
                 }
             }
 
-            // write flags
+            ushort genericParamRefId = 0xFFFF;
+
+            if (item.HasGenericParameters)
+            {
+                // no need to check if it's found
+                _context.GenericParamsTable.TryGetParameterId(item.GenericParameters.FirstOrDefault(), out genericParamRefId);
+            }
+
+            // FirstGenericParam
+            writer.WriteUInt16(genericParamRefId);
+
+            // GenericParamCount
+            writer.WriteByte((byte)item.GenericParameters.Count);
+
+            // Flags
             writer.WriteUInt16(
                 (ushort)GetFlags(
                     item,
                     _context.MethodDefinitionTable));
+
+            var writerEndPosition = writer.BaseStream.Position;
+
+            // ignore assert when not minimize
+            if (_context.MinimizeComplete)
+            {
+                Debug.Assert((writerEndPosition - writerStartPosition) == sizeOf_CLR_RECORD_TYPEDEF);
+            }
+        }
+
+        /// <summary>
+        /// Add a "fake" TypeDef as placeholder for an instanced generic type.
+        /// </summary>
+        /// <param name="typeReference"></param>
+        public void AddGenericInstanceType(TypeReference typeReference)
+        {
+            // drop namespace as it's already on the full name
+            // OK to use full name as type name to help comparison ahead
+            var genericType = new TypeDefinition(
+                "",
```

Comment:
very very very minimal
```suggestion
                string.Empty,
```